src/styles/main.scss

```diff
@@ -1,3 +1,74 @@
+$calendar-block-color: #eee;
+$border-color: #000;
+
+$gap: 1px;
+$border: 1px;
+$your-block-size: 100px;
+
+@function calendarSize() {
+  @return 7 * $your-block-size + $gap * 9;
+}
+
+@function block-size() {
+  @return $your-block-size - ($border * 1.6);
+}
+
+@for $index from 1 through 31 {
+  span:nth-of-type(#{$index}):before {
+    content: "#{$index}";
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    font-size: 30px;
+  }
+}
+
+@for $index from 28 through 31 {
+  .calendar--month-length--#{$index} {
+    .calendar__day:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+}
+
+$days:
+  "mon" 0,
+  "tue" calendarSize() - 6.08 * $your-block-size,
+  "wed" calendarSize() - 5.06 * $your-block-size,
+  "thu" calendarSize() - 4.05 * $your-block-size,
+  "fri" calendarSize() - 3.04 * $your-block-size,
+  "sat" calendarSize() - 2.02 * $your-block-size,
+  "sun" calendarSize() - 1.01 * $your-block-size;
```

Comment:
why do we need this float values? please check this screenshot, your items are not straight
<img width="820" alt="Screenshot 2023-01-18 at 11 13 42 AM" src="https://user-images.githubusercontent.com/60295799/213130916-74666527-3123-4ada-8880-67296f3a2d5a.png">